app/javascript/controllers/calendar_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["date"]
+
+  connect() {
+    this.dateTargets.forEach((date) => {
+      date.parentElement.addEventListener("click", this.toggleSelect)
+
+      if (date.hasAttribute("selected")) {
+        date.parentElement.setAttribute("selected", true)
+        date.parentElement.classList.add("bg-primary", "text-light")
+      }
+    })
+  }
+
+  toggleSelect(event) {
```

Comment:
since the `event.target` is the parent td box, the `disabled` attribute is added/removed on the child, which is the actual `hidden_field` form element that needs to be enabled/disabled to make through to the controller params.